src/jit/flowgraph.cpp

```diff
@@ -13681,13 +13681,19 @@ bool Compiler::fgOptimizeUncondBranchToSimpleCond(BasicBlock* block, BasicBlock*
         return false;
     }
 
+    if (block->IsLIR())
+    {
+        // NOTE: we do not currently hit this NYI because this function is only called when
+        // `fgUpdateFlowGraph` has been called with `doTailDuplication` set to true, and the
+        // backend always calls `fgUpdateFlowGraph` with `doTailDuplication` set to false.
+        NYI("fgOptimizeUncondBranchToSimpleCond for LIR");
+    }
+
```

Comment:
I think you should just assert(!block->IsLIR()) with this comment; no need for an NYI -- it's not so good to have NYI is shipping compiler code paths.